catroid/src/androidTest/java/org/catrobat/catroid/test/stage/InvalidSaveScreenshotTest.java

```diff
@@ -30,21 +30,34 @@
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
-import static org.catrobat.catroid.stage.StageListener.SCREENSHOT_AUTOMATIC_FILE_NAME;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
 import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.fail;
 
 @RunWith(AndroidJUnit4.class)
 public class InvalidSaveScreenshotTest {
 	private StageListener stageListener;
 
+	private static final String INVALID_SCREENSHOT_NAME = "   ";
+
 	@Before
 	public void setUp() throws Exception {
 		stageListener = new StageListener();
 	}
 
 	@Test
 	public void testInvalidScreenshotSave() {
+		CountDownLatch expectation = new CountDownLatch(1);
 		stageListener = new StageListener();
-		assertFalse(stageListener.saveScreenshot(null, SCREENSHOT_AUTOMATIC_FILE_NAME));
+		stageListener.requestTakingScreenshot(INVALID_SCREENSHOT_NAME,
```

Comment:
Unfortunately this tests nothing, because requestTakingScreenshot only sets some variables. 
Consider using Mockito afterwards to verify that the callback was executed.

Reply:
the assertion you are looking for is in line 55. This test was migrated to that version because of a crash when using `null` as screenshot name. I've added a few more invalid cases

Reply:
I get the assertion, but this callback method (with the assertion) wont be executed. You could call screenshotSaver.saveScreenshotAndNotify(..) instead.